2022q3/gcc.adoc

```diff
@@ -0,0 +1,31 @@
+=== GCC: New maintainer, GCC 12.2 and more
+
+Links: +
+link:https://gcc.gnu.org[GCC Project] URL: link:https://gcc.gnu.org[https://gcc.gnu.org] +
+link:https://gcc.gnu.org/gcc-11/[GCC 11 release series] URL: link:https://gcc.gnu.org/gcc-11/[https://gcc.gnu.org/gcc-11/]
+link:https://gcc.gnu.org/gcc-12/[GCC 12 release series] URL: link:https://gcc.gnu.org/gcc-12/[https://gcc.gnu.org/gcc-12/]
+
+Contact: <toolchain@FreeBSD.org> +
+Contact: Lorenzo Salvadore <salvadore@FreeBSD.org> +
+
+ * salvadore@ adopted all existing ports corresponding to supported versions of gcc, namely: lang/gcc10, lang/gcc11, lang/gcc11-devel, lang/gcc12, lang/gcc12-devel and lang/gcc13-devel.
+   At the moment -devel ports are updated weekly, unless a build failure makes it impossible.
+   Of course, in the latter case, the build failure is fixed and/or reported upstream as soon as possible.
+
+ * link:https://gcc.gnu.org/gcc-12/changes.html[GCC 12.2 has been released] and thus work has started to update the default GCC version to version 12.
+   Thank you very much to antoine@ who has already run the firt exp-run and to all the contributors, maintainers and committers involved in the process.
+   link:https://bugs.freebsd.org/bugzilla/show_bug.cgi?id=265948[https://bugs.freebsd.org/bugzilla/show_bug.cgi?id=2659548]
+
+ * Discussion about LTO keep going with many different point of views.
+   If interested, you can read the last contributions to the topic link:https://bugs.freebsd.org/bugzilla/show_bug.cgi?id=264949[here] and link:https://bugs.freebsd.org/bugzilla/show_bug.cgi?id=265254[here].
+   Reminder: LTO_BOOTSTRAP is an option enabled by default.
+   If you build the port on your machine and its resources consumption is not acceptable, disabling this option will get you a lighter compilation.
+
+ * jbeich@ submitted a patch to expose non-default -stdlib=libc++ support, which has been successfully committed to all relevant ports (gcc >= 11).
+   link: https://bugs.freebsd.org/bugzilla/show_bug.cgi?id=265962[https://bugs.freebsd.org/bugzilla/show_bug.cgi?id=265962]
+
+ * Help is still needed with these three changes to work through with upstream GCC (requires src expertise, not ports):
```

Comment:
Clarify: src as in the FreeBSD src repo/source tree, or as in GCC source?

Reply:
I think it is GCC source, but I am copy-pasting this part from gerald's last GCC quarterly report, so it is better to ask him.

 @GeraldPfeifer: Could you please confirm that I have understood correctly? Also, feel free to send me any hint on how to address those issues if you have any: since this is already at least the third quarter we add those lines, I would like to try to fix them.

Reply:
When originally writing this I was thinking of FreeBSD src committer skills (i.e., systems programming) over FreeBSD ports committer skills (i.e., packaging), though of course the lines a blury.

What I meant to relay is that it requires understanding and hacking compilers, in particular GCC, as opposed to regular ports work.

Reply:
Thanks Gerald. I have already fixed the report clarifying this point in commit e7f5cfa6a1abd05fc015ef7f8351509afc65af6c and it goes in the direction you have just outlined. Please feel free to suggest improvements if you want.